addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -55,15 +56,6 @@ const FormEditor = options.Class.extend({
         }
         return field.records;
     },
-    /**
-     * Generates a new ID.
-     *
-     * @private
-     * @returns {string} The new ID
-     */
-    _generateUniqueID() {
-        return Math.random().toString(36).substring(2, 15);
```

Comment:
I had no preference between the 2 implementation, so I kept this one, but indeed the parameter should be optional/default to smth

Reply:
We need the new implemenation because in the _notifyCurrentState function of the listUserValueWidget, there is some code that tries to parse the id as an Integer if the id is "1234azerty" the "azerty" will be lost and it will cause problems. So we need an id generation composed only of characters (the new implementation).

On top of that, it seems to me that the fact that an HTML element that has an ID that doesn't start with a letter can cause problems (for CSS selectors for example, ...)

I agree on the parameters, we can remove it and return each time an id of the same size (30?)

Reply:
Mmh indeed an ID should start by a letter (arguably it should be a fix in stable) but it can contain one even if you try to parse it as integer for some strange reason.

I would have gone with
```javascript
return `o${Math.random().toString(36).substring(2,15)}`;
```

That solves the letter problem and keep a short and efficient implementation. And you don't need 30 characters, when we made the form implementation we choose that "small" size because the risk of conflict is already incredibly small and without noising the DOM too much. But all of that is details so whatever :wink: I just wondered why we replaced a one line easy-to-understand implementation by a longer one which contains "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijkklmnopqrstxyz" :grimacing: